### PR TITLE
feat: add upstream command

### PR DESCRIPTION
**Pull request by wtrocki**
The basic idea for setup command.

I think getting started is quite bad as init command is using anilist.
Setup command will allow dynamic generation of config. 

## How this works

./graphql-gw setup https://localhost:8040 

Will produce following content:

```
#
# Configure the host and port the service will listen on
listen: localhost:8080

#
# Configure the GraphQL upstream servers you will be accessing
upstreams: 
   localhost:
      url: https://localhost:8081

types:
      - name: Query
        actions:
          # mounts all the fields of the root query onto the Query type
          - type: mount
            upstream: localhost
            query: query {}

      - name: Mutation
        actions:
          # mounts all the fields of the root query onto the Mutation type
          - type: mount
            upstream: localhost
            query: mutation {}

      - name: Subscription
        actions:
          # mounts all the fields of the root query onto the Subscription type
          - type: mount
            upstream: localhost
            query: subscription {}
        
```

### CODE REVIEW

**Comment by wtrocki** [comment]
As follow up we can actually perform check for the schemas and suggest renames etc.

**Comment by chirino** [comment]
Graphql servers don't have to implement a root mutation or a subscription.  Will this setup still work if it's used against one of those?

**Comment by wtrocki** [comment]
There is no issue in the setup. It is more about how auto import works. 
In this setup snippet we just say that we want to get all mutations, queries and subscriptions. Even queries are not enforced in the last iteration of the spec. 

So users will be able to remove those if they do not have subscriptions, even if they do not remove it they should end up with something like

```
type Mutation {}
type Subscription {}

```

Which is fine

**Comment by wtrocki** [comment]
As follow up we can work on some service feedback mechanism that it will fix it.

1. The user gives lists of upstreams
2. We download them and perform analysis of the schema
3. We provide wizard for type checks collisions and import root query, subscription and mutation if they are present.

**Comment by chirino** [comment]
I like the idea about downloading and inspecting the schema so we generate the right config for the given upstream.  Plus it allows us to validate it's a valid upstream before adding it to the config file.  Also given that users might want to do setup on multiple upstreams, maybe this `setup` command should instead become `add upstream` command.  Where we edit the existing config so they users can do it multiple times.

**Comment by wtrocki** [comment]
Yes. I will refactor this PR to contain new command

**Comment by wtrocki** [comment]
I have done some progress on this but still working on obtaining info from upstream

File: internal/cmd/upstream/cmd.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+	url, err := url.Parse(UpstreamURL)
+	if err != nil {
+		log.Fatalf(vebosityFmt, err)
+		panic(err)
+	}
+
+	log.Printf(`Contacting upstream %s`, url)
+
+	// TODO refactor loadEndpointSchema
```
Comment (wtrocki) [inline internal/cmd/upstream/cmd.go:66]: So the idea here is that I will look for Query, Mutation and Subscription types in the remote schema and mount them here

**Comment by wtrocki** [comment]
I actually had some unpublished experiments. haven't really got chance to go back to this. Sorry

**Comment by wtrocki** [comment]
Command is not really functional. if you manage to get this done please ping me on PR

**Pull request merged**